internal/engine/session/conv.go

```diff
@@ -44,15 +44,10 @@ func k8sRuntimeTarget(mt *store.ManifestTarget) *session.Target {
 
 	target := &session.Target{
 		Name:      fmt.Sprintf("%s:runtime", mt.Manifest.Name.String()),
+		Type:      k8sTargetType(mt),
 		Resources: []string{mt.Manifest.Name.String()},
 	}
 
-	if mt.Manifest.IsK8s() && mt.Manifest.K8sTarget().HasJob() {
```

Comment:
fwiw: i suspect what's eventually going to happen is that:
1) we're going to add PodReadinessMode to KubenetesApply/KubernetesDiscovery Spec
2) There will be a PodReadinessMode="auto" that infers based on YAML
3) KubernetesApplyStatus will reflect whether the inferred readiness is Server or Job
4) KuberentesApply will have a metav1.Condition{Name: "Ready"} that reflects readiness

and so a lot of this inference will be in the API

Reply:
That's what I was imagining, so glad we're on the same page.

(The short-term impact here is not my favorite...as a user, I would generally assume a `type` field would be immutable. But since the data within any given version of a `Session` obj will be consistent, it doesn't actually introduce any races so 🤷)